blocks/gmo-campaign-list/AssetsDatasource.js

```diff
@@ -0,0 +1,99 @@
+import { getBearerToken } from '../../scripts/security.js';
+
+import {
+  getAssetHandlerApiKey,
+  getDeliveryEnvironment,
+  getBackendApiKey,
+  getSearchIndex,
+  initDeliveryEnvironment,
+  getOptimizedDeliveryUrl
+} from '../../scripts/polaris.js';
+
+import { getAdminConfig } from '../../scripts/site-config.js';
+
+import { logError } from '../../scripts/scripts.js';
+
+async function getRequestHeadersSearchAssets() {
+  const token = await getBearerToken();
+  return {
+    'Content-Type': 'application/json',
+    'x-api-key': getBackendApiKey(),
+    Authorization: token,
+    'x-adobe-accept-experimental': '1',
+    'x-adp-request': 'search',
+  };
+}
+
+const getFilters = () => {
+  const currentDate = new Date();
+  const currentEpoch = Math.floor(currentDate.getTime() / 1000);
+  // Algolia does not support filters based on mixed types; for example boolean & numeric field types
+  // is_pur-expirationDate is a boolean type; but aloglia's engine treats boolean false as 0
+  // so we can use that to our advantage for numberic filters
+  return `is_pur-expirationDate = 0 OR pur-expirationDate > ${currentEpoch}`;
+};
+
+
+/**
+ * Search Asset for  programName and campaignName parameters.
+ *
+ * @param {string} - Program Name.
+ * @param {string} - Campaign Name.
+ * @returns {Image <object>} Resolves with image object.
+ * @throws {Error} If an HTTP error or network error occurs.
+ */
+
+export async function searchAsset(programName, campaignName) {
+  const adminConfig = await getAdminConfig();
+  const deliveryURL = await initDeliveryEnvironment();
+  const searchURL = `${deliveryURL}/adobe/assets/search`
```

Comment:
use `createSearchEndpoint` function of `scripts.js` to get the search endpoint url